api/src/main/java/org/commonjava/indy/cluster/IndyNodeHolder.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) 2013~2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.cluster;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+@ApplicationScoped
+public class IndyNodeHolder
+{
+    public static final Logger logger = LoggerFactory.getLogger( IndyNodeHolder.class );
+
+    public IndyNode getIndyNode()
+    {
+        //TODO: here is just a temporary solution to get a local indy node indicator. Will think more about this way future
+        try
+        {
+            return new IndyNode( InetAddress.getLocalHost().getHostName() );
```

Comment:
In Ying's PR, we talked about the way to get a proper hostname. Briefly, we can cat /etc/hostname, read $hostname sys var, or use above one. I don't know which one is better but we need to stick to one approach. CC @yma88 @jdcasey

Reply:
I think we need to alter the configuration-reader to allow expressions like `${env.HOSTNAME}` as that's probably the cheapest / most reliable way to detect it. getLocalHost().getHostName() may rely on one or both of /etc/hosts and /etc/hostname, which might not reflect a hostname change that was made via bootup script.